setup.py

```diff
@@ -196,20 +216,33 @@ def build_extension(self, ext):
         if DEVTOOLS == "MsDev":
             _xbuild = '"{0}"'.format(self._find_msbuild_tool("msbuild.exe"))
             _config = "{0}Win".format(CONFIG)
-
+            _solution_file = 'pythonnet.sln'
+            _custom_define_constants = False
+        elif DEVTOOLS == "MsDev15":
+            _xbuild = '"{0}"'.format(self._find_msbuild_tool_15()) 
+            _config = "{0}Win".format(CONFIG)
+            _solution_file = 'pythonnet.15.sln'
+            _custom_define_constants = True
         elif DEVTOOLS == "Mono":
-            _xbuild = "xbuild"
+            _xbuild = 'xbuild'
             _config = "{0}Mono".format(CONFIG)
+            _solution_file = 'pythonnet.sln'
+            _custom_define_constants = False
+        elif DEVTOOLS == "dotnet":
+            _xbuild = 'dotnet msbuild'
+            _config = "{0}Mono".format(CONFIG)
```

Comment:
why dotnet DEVTOOLS corresponds to Mono _config?

Reply:
The first problem:
DebugMonoPY3, ReleaseMono, etc. build configuration is a historically names from times where nothing than mono had exists for the linux.
Now it is more correctly to name DebugLinuxPY3, ReleaseLinux, etc.

http://cs5.pikabu.ru/images/big_size_comm/2015-03_2/14257150104254.jpg

But we "cannot just rename" build configurations without affecting users.
We should introduce new build configurations (DebugLinuxPY3, ReleaseLinux), give a time to users to migrate to new names, and than remove old names DebugMonoPY3, ReleaseMono, etc.

The second problem:
We cannot use dotnet build command under windows due to DllExport project msbuild targets. We needs to use msbuild >=15 under windows 

The result:

DEVTOOLS=dotnet ==> linux (--xplat) === ReleaseLinuxPY3 <== ReleaseMonoPY3

So currently we have DOVTOOLS=dotnet <===> ReleaseMonoPY3

If I am wrong in somewhere in my conclusions, lets improve something.